### DISCUSSION TITLE
Azure.AKS.MinNodeCount Enhancement

### DISCUSSION THREAD

**Discussion by paulmcclbg** [Azure/PSRule.Rules.Azure#2671] [Ideas]
The above rule checks for the first array member has a min count of 3

field: 'Properties.agentPoolProfiles[0].count'

We have clusters with more than one agent pool and the rule is failing because the first pool in the array only has 2 nodes, whereas the second pool in the array does have 3.

We need the ability to narrow the rule to a specific pool.

**Comment by BernieWhite** [comment]
@paulmcclbg Good call out. This should be improved.

Is the first node pool with 2 nodes the system node pool? and the second a user node pool?

Would it be acceptable to have a minimum count for `system` node pools. and a separate minimum count for `user` node pools.

**Reply by paulmcclbg** [reply]
@BernieWhite AKS not my area but both pools have a property called "mode" containing a value "System". In our case I think we would need to differentiate on the "name" property.

**Reply by BernieWhite** [reply]
Tracking as #2683

**Reply by BernieWhite** [reply]
This shipped in v1.34.0.
